src/Microsoft.Diagnostics.Runtime/src/Builders/RuntimeBuilder.cs

```diff
@@ -655,6 +657,140 @@ private IEnumerable<ClrHandle> EnumerateHandleTable(ClrRuntime runtime, HandleDa
             }
         }
 
+        IEnumerable<ClrNativeHeapInfo> IRuntimeHelpers.EnumerateClrNativeHeaps(ulong systemDomainAddress, ulong sharedDomainAddress)
+        {
+            List<ClrNativeHeapInfo> heaps = new();
+
+            // Enumerate JIT code heaps.
+            foreach (JitManagerInfo jitMgr in _sos.GetJitManagers())
+            {
+                foreach (var mem in _sos.GetCodeHeapList(jitMgr.Address))
+                {
+                    if (mem.Type == CodeHeapType.Loader)
+                    {
+                        if (TraverseLoaderHeap(mem.Address, SOSDac13.LoaderHeapKind.LoaderHeapKindExplicitControl, (address, size, _) => heaps.Add(new ClrNativeHeapInfo(address, SanitizeSize(size), NativeHeapKind.LoaderCodeHeap))))
```

Comment:
Wouldn't it be more efficient to create an enumerable out of the traversing code? That would avoid eagerly evaluating the data and holding onto it for the entirety of the enumeration.